### PR TITLE
Package/cloverleaf

### PR DESCRIPTION
**Pull request by homerdin**
Adding CloverLeaf Package

### CODE REVIEW

File: var/spack/repos/builtin/packages/cloverleaf/package.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    homepage = "http://uk-mac.github.io/CloverLeaf"
+    url      = "http://mantevo.org/downloads/releaseTarballs/miniapps/CloverLeaf/CloverLeaf-1.1.tar.gz"
+
+    version('1.1', '65652b30a64eb237ec844a6fdd4cd518')
+
+    variant('build', default='ref', description='Type of Parallelism Build',
+            values=('cuda', 'mpi', 'openacc_cray', 'openmp', 'ref', 'serial'))
+
+    depends_on('mpi')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cloverleaf/package.py]: Should this be `when='build=mpi'`?

File: var/spack/repos/builtin/packages/cloverleaf/package.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            build = 'OpenMP'
+        elif 'build=serial' in self.spec:
+            build = 'Serial'
+
+        return build
+
+    @property
+    def build_targets(self):
+        targets = [
+            'MPI_COMPILER={0}'.format(self.spec['mpi'].mpifc),
```
Comment (davydden) [inline var/spack/repos/builtin/packages/cloverleaf/package.py]: if `mpi` dependency is conditional, this should also be conditional

File: var/spack/repos/builtin/packages/cloverleaf/package.py (hunk 1 above)
Comment (homerdin) [inline var/spack/repos/builtin/packages/cloverleaf/package.py]: Looked into it and they are using the mpi wrapper in all builds to abstract it.  Only the 'cuda', 'mpi', 'openacc_cray' and 'ref' builds depend on mpi.  I am making the changes.  Is there a more concise way of expressing this besides one at a time?
```
    depends_on('mpi', when='build=cuda')
    depends_on('mpi', when='build=mpi_only')
    depends_on('mpi', when='build=openacc_cray')
    depends_on('mpi', when='build=ref')
```

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cloverleaf/package.py]: Nope, that looks right to me.

**Pull request merged**